### PR TITLE
Build for web and dockerfile

### PR DESCRIPTION
**Pull request by gwbischof**
This PR gets the build for web working, and adds a dockerfile to build the web image.
The Platform class which is used to check the platform it is on is not supported by web, so `kIsWeb` is used to check if the platform is web.
See discussion on old PR: https://github.com/thunder-app/thunder/pull/927

I also have a github workflow to build the image on tag/release.  But I think thunder-app GHCR needs to be setup first.
There are a couple things not working quite right on web still, I can fix those up in follow up PRs.

I followed the instructions here to add the sqflight binaries: https://pub.dev/packages/sqflite_common_ffi_web#setup-binaries

### CODE REVIEW

File: scripts/docker-build-web.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+D=$(date +'%Y%m%d.%H%M%S%3N')
```
[the remark this answers is not in the document]
Comment (gwbischof) [reply scripts/docker-build-web.sh]: I just removed the date, it was left over from copy-pasting from an existing script.

File: docker/DockerfileWeb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+RUN flutter upgrade
+
+# Copy files to container and build
+RUN mkdir /app/
+COPY . /app/
+WORKDIR /app/
+RUN flutter build web --no-tree-shake-icons
+
+# Stage 2 - Create the run-time image
+FROM nginx:1.21.1-alpine
```
[the remark this answers is not in the document]
Comment (gwbischof) [reply docker/DockerfileWeb]: I just updated it and tested.

File: scripts/docker-build-web.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
```
[the remark this answers is not in the document]
Comment (micahmo) [reply scripts/docker-build-web.sh:3]: Ah ok cool! I was just curious!

**Comment by gwbischof** [comment]
Merge please :smile:

**Pull request merged**